client/src/js/components/bhEntityTypeSelect/bhEntityTypeSelect.js

```diff
@@ -0,0 +1,44 @@
+angular.module('bhima.components')
+  .component('bhEntityTypeSelect', {
+    templateUrl : 'js/components/bhEntityTypeSelect/bhEntityTypeSelect.tmpl.html',
+    controller  : EntityTypeSelectController,
+    transclude  : true,
+    bindings    : {
+      id               : '<',
+      onSelectCallback : '&',
+      required         : '<?',
+      label            : '@',
+    },
+  });
+
+EntityTypeSelectController.$inject = ['EntityService', 'NotifyService', '$translate'];
+
+/**
+ * EntityType selection component
+ */
+function EntityTypeSelectController(Entities, Notify, $translate) {
+  const $ctrl = this;
+
+  $ctrl.$onInit = function onInit() {
+    $ctrl.loading = true;
+
+    // load all entity types
+    Entities.types.read(null)
+      .then((types) => {
+        $ctrl.types = types.map(type => {
+          type.hrLabel = $translate.instant(type.translation_key || type.label);
+          return type;
+        });
+
+      })
+      .catch(Notify.handleError)
+      .finally(() => {
+        $ctrl.loading = false;
```

Comment:
I'm not sure anything here uses the `loading` indicator.  If we don't use it, it should be removed.